src/tox/session/commands/run/parallel.py

```diff
@@ -75,7 +75,7 @@ def collect_process(process):
                 elif status is not None:
                     outcome = spinner.fail
                 outcome(env_name)
-                if print_out and output is not None:
+                if vars().get("print_out", None) and output is not None:
```

Comment:
I don't think this is a correct solution, probably just need to move the definition of `print_out` outside of the try block

Reply:
Sure. The hard question is about the missing warning, which causes that specific test to fail.